### PR TITLE
CLN use pd.Timestamp for single element in converter

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Only a single element is being parsed, and there's no way for a user to pass any options (like `dayfirst`) through `ax.set_xlim`, so might as well use `pd.Timestamp`.

Otherwise with https://github.com/pandas-dev/pandas/pull/49024 users might get a warning they wouldn't be able to do anything about

Example of test that hits this: `pytest pandas/tests/plotting/test_datetimelike.py -k test_time_change_xlim`

### CODE REVIEW

File: pandas/plotting/_matplotlib/converter.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def _to_ordinalf(tm: pydt.time) -> float:
 
 def time2num(d):
     if isinstance(d, str):
-        parsed = tools.to_datetime(d)
+        parsed = Timestamp(d)
```
Comment (phofl) [inline pandas/plotting/_matplotlib/converter.py:155]: This probably changes error messages? If yes, should add a whatsnew

Comment (jbrockmendel) [reply pandas/plotting/_matplotlib/converter.py:155]: i think this also makes the check/raise on the next two lines unnecesary

Comment (MarcoGorelli) [reply pandas/plotting/_matplotlib/converter.py:155]: yeah good call, looks like they're uncovered as it is anyway https://app.codecov.io/gh/pandas-dev/pandas/blob/main/pandas/plotting/_matplotlib/converter.py

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @MarcoGorelli

**Pull request merged**